app/rest/proxy/avatar_store.go

```diff
@@ -56,7 +65,22 @@ func (fs *FSAvatarStore) Put(userID string, reader io.Reader) (avatar string, er
 		}
 	}()
 
-	if _, err = io.Copy(fh, reader); err != nil {
+	img, err := ioutil.ReadAll(reader)
```

Comment:
I think this can be done directly on reader, without this ReadAll putting potentially enormous []byte to memory